src/lib.rs

```diff
@@ -1029,6 +1029,15 @@ struct MatchTreeLevel<'a> {
     branches: Vec<(ByteSet, LevelBranch<'a>)>,
 }
 
+/// A bundle of follow-sets with an externally significant index, e.g. into an array of decoders
+///
+/// While the choice-points along any given descent of a [`MatchTree`] are fully informed
+/// by sets of mutually disjoint [`ByteSet`]s, the superposition of choice-points across
+/// all descents of a fixed depth (i.e. a [`MatchTreeLevel`]) will almost invariably encompass
+/// multiple original choice-points with non-disjoint [`ByteSet`]s. In such cases, it may be necessary
+/// to create multiple virtual branches that represent a deterministic superposition in the otherwise
+/// nondeterministic single-byte-descent at that level. In such cases, we cross-reference an original index
+/// across all entries stemming from the same [MatchTreeStep].
```

Comment:
I'm not sure I can understand this comment, and it might be better suited to external documentation.